WISE-Community/WISE-Client
# feat(Dynamic Prompt): Evaluate rules using Peer Group work

## Changes

- FeedbackRuleEvaluator.getFeedbackRule() can now accept a single CRaterResponse or an array of CRaterResponses
- Dynamic prompt can figure out which dynamic prompt to display based on work and scores from the members of a Peer Chat group

## Test

- Make sure the FeedbackRuleEvaluator works when it receives one CRaterResponse (when you do not specify the dynamicPrompt.peerGroupingTag)
- Make sure the FeedbackRuleEvaluator works when it receives an array of CRaterResponses (when you specify the dynamicPrompt.peerGroupingTag)

Here is the Peer Chat component content I used for testing (when peerGroupingTag is specified).
```
{
    "id": "gkvyzuils1",
    "type": "PeerChat",
    "prompt": "",
    "showSaveButton": false,
    "showSubmitButton": false,
    "peerGroupingTag": "847zgyqp1p",
    "questionBank": [],
    "dynamicPrompt": {
        "enabled": true,
        "peerGroupingTag": "847zgyqp1p",
        "prePrompt": "This is the pre prompt",
        "postPrompt": "This is the post prompt",
        "referenceComponent": {
            "nodeId": "node49",
            "componentId": "2aabvbcukq"
        },
        "rules": [
            {
                "id": "4y3h9u0sdh",
                "expression": "isSecondToLastSubmit",
                "prompt": "Prompt when you still have a submit to use"
            },
            {
                "id": "df92gk754j",
                "expression": "2 && 3",
                "prompt": "Prompt when you got idea 2 and idea 3"
            },
            {
                "id": "7j2is1360s",
                "expression": "2 || 3",
                "prompt": "Prompt when you got idea 2 or idea 3"
            },
            {
                "id": "votbik4ncz",
                "expression": "hasKIScore(5)",
                "prompt": "Prompt when you got 5"
            },
            {
                "id": "hwp6sxf8vr",
                "expression": "hasKIScore(4)",
                "prompt": "Prompt when you got 4"
            },
            {
                "id": "swxzc0j7q0",
                "expression": "hasKIScore(3)",
                "prompt": "Prompt when you got 3"
            },
            {
                "id": "ebtgds3b1r",
                "expression": "hasKIScore(2)",
                "prompt": "Prompt when you got 2"
            },
            {
                "id": "z4yu37jcis",
                "expression": "hasKIScore(1)",
                "prompt": "Prompt when you got 1"
            },
            {
                "id": "isDefault",
                "expression": "isDefault",
                "prompt": "Default prompt"
            }
        ]
    }
}
```

#840
